### DISCUSSION TITLE
Collection with GPU_IVF_FLAT index

### DISCUSSION THREAD

**Discussion by vtnhan1309** [milvus-io/milvus#35488] [Q&A and General discussion] [answered]
When I loaded a collection with 10M vectors, I noticed that Milvus took up 40GB VRAM at total. What happen if we cannot provide enough VRAM in this case ? (ex: we only have a GPU with 24GB VRAM) ?

**Comment by yhmo** [comment]
10M vectors with 512 dim, the original data size is 512*4Byte * 10M = 20GB.

The optional GPU index types: GPU_CAGRA, GPU_IVF_FLAT, GPU_IVF_PQ, and GPU_BRUTE_FORCE

The size of GPU_CAGRA/GPU_IVF_FLAT index is a bit larger than the original size. High recall rate( > 95%).
The size of GPU_BRUTE_FORCE index is equal to the original size. 100% recall rate.
The size of GPU_IVF_PQ is about 25%~30% of the original size. Middle recall rate ( > 90%).

Currently, there is an expansion rate(nearly 2) when loading the GPU index into GPU memory. So, 24GB GPU memory is insufficient to load GPU_CAGRA/GPU_IVF_FLAT/GPU_BRUTE_FORCE. 
The GPU_IVF_PQ is much smaller than others, it should be ok to work with 24GB VRAM.

**Comment by xiaofan-luan** [comment, marked as answer]
> on with 10M

1. it depends on the dimension, for 768dim yes you can not hold everything
2. you can use multiple card or distributed solution, we don't required to put everything into single machine
